src/common/buf.c

```diff
@@ -5,6 +5,15 @@
 #include "common/buf.h"
 #include "common/mem.h"
 
+static void
+buf_replace_by(struct buf *dst, struct buf *src)
```

Comment:
Consider calling it `buf_take()` or `buf_steal()`. Common when transferring ownership of heap-allocated memory.

Reply:
Personally I like `buf_replace_by()` better, especially as it somewhat implies what is replaced by what (e.g. first arg is the buffer this API targets / `self` and second then must be the src). But its just some internal static function so also fine changing it to something else if you insist.

Reply:
I would vote for `buf_move` since this is analogous to move-assignment from C++.